### PR TITLE
feat(locales.pl): add pl translation for gender and binary gender

### PR DESCRIPTION
**Pull request by hankucz**

### CODE REVIEW

**Review by Shinigami92 — DISMISSED** [review]

File: src/locales/pl/name/gender.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  'Osoba bigender',
+  'Osoba gender fluid',
+  'Osoba interpłciowa',
+  'Osoba niebinarna',
+  'Osoba pangender',
+  'Osoba polygender',
+  'Osoba trans',
+  'Trankseksualna kobieta',
+  'Trankseksualna osoba',
+  'Transeksualny mężczyzna',
```
Comment (ST-DDT) [inline src/locales/pl/name/gender.ts:25]: Just to confirm is this difference in spelling intentional? Google translate seems a bit confused here.

**Review by ST-DDT — APPROVED** [review]

File: src/locales/pl/name/gender.ts (hunk 1 above)
Comment (pkuczynski) [reply src/locales/pl/name/gender.ts:25]: No, that's error! Very good catch! :)

Comment (hankucz) [reply src/locales/pl/name/gender.ts:25]: Thank you for noticing, it is an error. Just fixed it :)

**Review by pkuczynski — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Please remove the unrelated commits/reverts from this PR by either rebasing/squashing and filtering the commits.

**Comment by pkuczynski** [comment]
Whats the difference? We will squash on merge anyway, right?

File: .eslintrc.js [hunk 2]
```diff
@@ -51,15 +51,6 @@ module.exports = defineConfig({
...
     '@typescript-eslint/unbound-method': 'off',
-    '@typescript-eslint/naming-convention': [
-      'error',
-      {
-        format: ['PascalCase'],
-        selector: ['class', 'interface', 'typeAlias', 'enumMember'],
-        leadingUnderscore: 'forbid',
-        trailingUnderscore: 'forbid',
-      },
-    ],
```
Comment (Shinigami92) [inline .eslintrc.js:62]: This needs to be reverted

File: src/modules/date/index.ts [hunk 3]
```diff
@@ -20,8 +20,6 @@ function toDate(date?: string | Date | number): Date {
 /**
  * Module to generate dates.
  */
-// disabled until modules are renamed to something with a suffix
-// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (Shinigami92) [inline src/modules/date/index.ts:24]: This needs to be reverted

File: src/modules/name/index.ts [hunk 4]
```diff
@@ -1,11 +1,8 @@
 import type { Faker } from '../..';
 import { deprecated } from '../../internal/deprecated';
 
-// disabled until renamed to Sex
 export enum Gender {
-  // eslint-disable-next-line @typescript-eslint/naming-convention
   female = 'female',
-  // eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (Shinigami92) [inline src/modules/name/index.ts:8]: This needs to be reverted

**Review by Shinigami92 — DISMISSED** [review]

**Comment by hankucz** [comment]
All unnecessary changes have been reverted. @ST-DDT @Shinigami92 could you please re-review and merge?

**Review by pkuczynski — APPROVED** [review]

**Comment by xDivisionByZerox** [comment]
> The current build failure is the timeout in `test/scripts/apidoc/examplesAndDeprecations.spec.ts`, which is not related to my PR. I am not sure if I have to do anything about it?

I opened an issue for it (#1193) since I struggle with it as well.

**Review by xDivisionByZerox — APPROVED** [review]

**Review by pkuczynski — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**